logic/atrium-logic-jvm/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultPathAssertions.kt

```diff
@@ -194,4 +193,32 @@ class DefaultPathAssertions : PathAssertions {
                     container.resolve(entry).collect { _logicAppend { exists(NOFOLLOW_LINKS) } }
                 }
         ).build()
+
+    override fun <T : Path> isEmptyDirectory(container: AssertionContainer<T>): Assertion {
+        val isDirectory = container.isDirectory()
+        if (isDirectory.holds()) {
+            return container.changeSubject.unreported {
+                Pair(it, it.runCatchingIo { Files.newDirectoryStream(it).use { stream -> stream.firstOrNull() } })
+            }.let { expectResult ->
+                    assertionBuilder.descriptive.withTest(expectResult)
+                    { it.second is Success && (it.second as Success<Path?>).value == null }
+                        .withFailureHintBasedOnDefinedSubject(expectResult) {
+                            explainForResolvedLink(it.first) { realPath ->
+                                when (it.second) {
+                                    is Success ->
+                                        assertionBuilder.descriptive.failing
+                                            .withDescriptionAndRepresentation(
+                                                DIRECTORY_CONTAINS,
+                                                (it.second as Success<Path?>).value!!
```

Comment:
I think this line could be simplified to:

```suggestion
                                                (it.second as Success<Path>).value
```